src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,623 @@
+
+        HandleCommand<Commands::ModifyPushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleModifyPushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::SetTransportStatus::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Setting Push Transport Status");
+
+        HandleCommand<Commands::SetTransportStatus::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleSetTransportStatus(ctx, commandData); });
+
+        break;
+
+    case Commands::ManuallyTriggerTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Manually Triggered Push Transport");
+
+        HandleCommand<Commands::ManuallyTriggerTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleManuallyTriggerTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::FindTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Finding Push Transport");
+
+        HandleCommand<Commands::FindTransport::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleFindTransport(ctx, commandData); });
+
+        break;
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+TransportConfigurationStructWithFabricIndex *
+PushAvStreamTransportServer::FindStreamTransportConnection(const uint16_t connectionID)
+{
+    for (auto & transportConnection : mCurrentConnections)
+    {
+        if (transportConnection.transportConfiguration.connectionID == connectionID)
+            return &transportConnection;
+    }
+    return nullptr;
+}
+
+uint16_t PushAvStreamTransportServer::GenerateConnectionID()
+{
+    static uint16_t lastAssignedConnectionID = 0;
+    do
+    {
+        uint16_t nextConnectionID;
+        if (lastAssignedConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+        {
+            nextConnectionID = 0;
+        }
+        else
+        {
+            nextConnectionID = static_cast<uint16_t>(lastAssignedConnectionID + 1);
+        }
+        lastAssignedConnectionID = nextConnectionID;
+        if (FindStreamTransportConnection(nextConnectionID) == nullptr)
+        {
+            return nextConnectionID;
+        }
+    } while (true);
+}
+
+void PushAvStreamTransportServer::HandleAllocatePushTransport(HandlerContext & ctx,
+                                                              const Commands::AllocatePushTransport::DecodableType & commandData)
+{
+    Commands::AllocatePushTransportResponse::Type response;
+    auto & transportOptions = commandData.transportOptions;
+
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(transportOptions.endpointID, TlsCertificateManagement::Id,
```

Comment:
But also, why are we treating a `TLSEndpointID` as an `endpoint-no`?  They're not the same thing.